Constructor change for Eidos

I see in the Indra code that someone is using
```python
eidos = autoclass(eidos_package + '.EidosSystem')
es = eidos(autoclass('java.lang.Object')())
```
to construct an EidosSystem.

This seems to make use of the EidosSystem special constructor
```scala
def this(x: Object) = this() // Dummy constructor crucial for Python integration
```
that might have been added becuase other languages cannot deal with the default value specified in the main constructor.
```scala
class EidosSystem(val config: Config = EidosSystem.defaultConfig) {
```
That extra constructor with the Object argument is starting to interfere with the internal calls in Eidos and it probably isn't even necessary.  I believe but am not sure that it could have been expressed previously as
```scala
class EidosSystem(val config: Config) {
  def this() = this(EidosSystem.defaultConfig)
```
and not required the special argument in Python at all.  I've made a branch of Eidos, `kwalcock-emptyConstructor` that can be used to test this.  Can someone there hook INDRA up to it to see if it works?

If it does, I'll merge that change and then afterwards move to more complicated constructors that account for the options we're needing to add, but keep that empty constructor rather than the Object one.  Thanks!

I'll write this down in case it looks familiar to anybody.  The problem seems to be in the constructor of the `org.apache.lucene.index.DirectoryReader`.  It complains
```
java.lang.ExceptionInInitializerError
	at org.apache.lucene.codecs.Codec.forName(Codec.java:116)
	at org.apache.lucene.index.SegmentInfos.readCodec(SegmentInfos.java:424)
	at org.apache.lucene.index.SegmentInfos.readCommit(SegmentInfos.java:356)
	at org.apache.lucene.index.SegmentInfos.readCommit(SegmentInfos.java:288)
	at org.apache.lucene.index.StandardDirectoryReader$1.doBody(StandardDirectoryReader.java:57)
	at org.apache.lucene.index.StandardDirectoryReader$1.doBody(StandardDirectoryReader.java:54)
	at org.apache.lucene.index.SegmentInfos$FindSegmentsFile.run(SegmentInfos.java:692)
	at org.apache.lucene.index.StandardDirectoryReader.open(StandardDirectoryReader.java:77)
	at org.apache.lucene.index.DirectoryReader.open(DirectoryReader.java:63)
	at org.clulab.wm.eidos.context.GeoNormFinder$.fromConfig(GeoNormFinder.scala:98)
	at org.clulab.wm.eidos.extraction.Finder$.$anonfun$fromConfig$1(Finder.scala:21)
	at scala.collection.immutable.List.map(List.scala:287)
	at org.clulab.wm.eidos.extraction.Finder$.fromConfig(Finder.scala:17)
	at org.clulab.wm.eidos.EidosSystem$LoadableAttributes$.apply(EidosSystem.scala:95)
	at org.clulab.wm.eidos.EidosSystem.<init>(EidosSystem.scala:122)
	at org.clulab.wm.eidos.EidosSystem.<init>(EidosSystem.scala:26)
Caused by: java.lang.IllegalArgumentException: An SPI class of type org.apache.lucene.codecs.Codec with name 'Lucene62' does not exist.  You need to add the corresponding JAR file supporting this SPI to your classpath.  The current classpath supports the following names: []
	at org.apache.lucene.util.NamedSPILoader.lookup(NamedSPILoader.java:116)
	at org.apache.lucene.codecs.Codec$Holder.<clinit>(Codec.java:60)
	... 16 more
```
I added
```
"org.apache.lucene" % "lucene-codecs"           % luceneVer
```
to build.sbt and verified that the class file /org/apache/lucene/codecs/lucene62/Lucene62Codec.class is part of the assembled file.  The error message did not change.

I copied lucene-codecs-6.6.6.jar to the docker container and adjusted the CLASSPATH to include this jar file.  The error message improved to
```
  You need to add the corresponding JAR file supporting this SPI to your classpath.  The current classpath supports the following names: [SimpleText]
```
I wonder whether the assembled file is really correct, because I don't see anything in META-INF that sounds like this:
`
Lucene 4 supports now so called "Codecs" which are implementation classes for custom index encodings. E.g. you can exchange the default postings list by a custom format. Those implementations are classes that extend an abstract base class/interface to implement this functionality. To plug this into the Lucene index format, IndexReader and IndexWriter need the knowledge, which codec type was used to load the correct classes. This is done by the Java Service Provider interface (SPI), which is used quite often e.g. to load custom image formats, XML parsers, Charsets,... in the JDK. The codec implementation classes are listed in the JAR files META-INF section and then loaded by the code and register themselves through the codec manager. There is no custom class loading involved, it's all just plain Java. E.g. Charset.forName("ISO-8859-1") in the JDK does exactly the same: It scans META-INF for Charset files, loads all charsets and looks them up by name. You can use this to e.g. plug in ICU4J to have more charsets than in the default JDK. 
`

--- a/indra/sources/eidos/reader.py
+++ b/indra/sources/eidos/reader.py
@@ -54,7 +54,7 @@ class EidosReader(object):
     def initialize_reader(self):
         """Instantiate the Eidos reader attribute of this reader."""
         eidos = autoclass(eidos_package + '.EidosSystem')
-        self.eidos_reader = eidos(autoclass('java.lang.Object')())
+        self.eidos_reader = eidos()
 
     def reground_texts(self, texts, yaml_str=None):
         if self.eidos_reader is None:

--- a/indra/sources/eidos/reader.py
+++ b/indra/sources/eidos/reader.py
@@ -101,10 +101,9 @@ class EidosReader(object):
 
         annot_doc = self.eidos_reader.extractFromText(
             text,
-            True, # keep text
-            False, # CAG-relevant only
-            default_arg(today), # doc creation time
-            default_arg(fname) # file name
+            False,  # CAG-relevant only
+            default_arg(today),  # doc creation time
+            default_arg(fname)  # file name
             )
         if format == 'json':
             mentions = annot_doc.odinMentions()

